File: C++/Two-Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> hash;
        vector<int> res, cnt;
        for (int i = 0; i < nums.size(); i ++) {
        	if (nums[i] * 2 == target) {
        		cnt.push_back(i + 1);
        	} else {
        		hash[nums[i]] = i + 1;
        	}
        }
        if (cnt.size() >= 2) {
        	res.push_back(cnt[0]);
        	res.push_back(cnt[1]);
        	return res;
        }
        for (int i = 0; i < nums.size(); i ++) {
        	if (hash.find(target - nums[i]) != hash.end()) {
        		int temp = hash[target - nums[i]];
        		res.push_back(min(temp, i + 1));
        		res.push_back(max(tmep, i + 1));
        		return res;
        	}
        }
        return res;
    }
};

File: Java/Two-Sum.java
public class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] res = new int[2];
        HashMap<Integer, Integer> hash = new HashMap<Integer, Integer>();
        int cur = 0;
        for (int i = 0; i < nums.length; i ++) {
            if (nums[i] == target / 2 && target % 2 == 0) {
                if (cur < 2) res[cur ++] = i + 1;
            }
            hash.put(nums[i], i);
        }
        if (cur >= 2) return res;
        for (int i = 0; i < nums.length; i ++) {
            if (nums[i] * 2 != target && hash.containsKey(target - nums[i])) {
                res[0] = hash.get(nums[i]) + 1;
                res[1] = hash.get(target - nums[i]) + 1;
            }
        }
        Arrays.sort(res);
        return res;
    }
}